GridTariffAPI.lib/Services/TariffQueryService.cs

```diff
@@ -53,60 +53,109 @@ public class TariffQueryService : ITariffQueryService
+            if (currentFixedPrices != null)
+            {
+                var priceLevels = AppendMeteringPointsToPriceLevels(meteringPointInformation, currentFixedPrices);
+                foreach (var priceLevel in priceLevels)
+                {
+                    gridTariff.MeteringPointsAndPriceLevels.Add(priceLevel);
+                }
+            }
             return gridTariff;
         }
 
-        public virtual async Task AppendMeteringPointsToPriceLevels(List<MeteringPointInformation> meteringPointInformation, GridTariffCollection gridTariff)
+        public List<MeteringPointsAndPriceLevels> MeteringPointsQueryOutsideUtcNow(
+            DateTimeOffset paramFromDate,
+            DateTimeOffset paramToDate,
+            List<MeteringPointInformation> meteringPointInformations)
         {
-            //todo should be some of filtering here (only current month?), thus no need for foreach fixedprices
-            foreach (var fixedPrice in gridTariff.GridTariff.TariffPrice.PriceInfo.FixedPrices)
+            var retVal = new List<MeteringPointsAndPriceLevels>();
+            if (paramFromDate > DateTimeOffset.UtcNow.Date || paramToDate < DateTimeOffset.UtcNow.Date)
             {
-                foreach (var fixedPriceLevel in fixedPrice.PriceLevel)
+                retVal = new List<MeteringPointsAndPriceLevels>();
+                var priceLevel = new MeteringPointsAndPriceLevels
                 {
-                    var meteringPointAndPriceLevel = await CheckPriceLevelForMeteringPoints(meteringPointInformation, fixedPrice, fixedPriceLevel);
-                    if (meteringPointAndPriceLevel != null)
-                    {
-                        gridTariff.MeteringPointsAndPriceLevels.Add(meteringPointAndPriceLevel);
-                    }
+                    MeteringPoints = new MeteringPoints()
+                };
+                foreach (var mp in meteringPointInformations)
+                {
+                    priceLevel.MeteringPoints.Add(new MeteringPointDetails() { MeteringPointId = mp.MeteringPointId });
                 }
+                retVal.Add(priceLevel);
             }
+            return retVal;
         }
 
-        public Task<MeteringPointsAndPriceLevels> CheckPriceLevelForMeteringPoints(List<MeteringPointInformation> meteringPointInformation, FixedPrices fixedPrice, FixedPriceLevel fixedPriceLevel)
+        public virtual List<MeteringPointsAndPriceLevels> AppendMeteringPointsToPriceLevels(List<MeteringPointInformation> meteringPointInformations, FixedPrices fixedPrices)
         {
-            MeteringPointsAndPriceLevels retVal = null;
+            var retVal = new List<MeteringPointsAndPriceLevels>();
+            //mp not missing maxConsumption
+            //mp missing maxConsumption and exactly one pricelevel
```

Comment:
Kan ikke se at vi håndterer mp missing maxConsumption and exactly one priceLevel.

Reply:
Jo, det er vel dekket av sjekken i CheckPriceLevelForMeteringPoints på "!fixedPriceLevel.ValueMin.HasValue && !fixedPriceLevel.ValueMax.HasValue".

Reply:
Skal nå være mer lettlest.